formatter.go

```diff
@@ -30,7 +31,7 @@ type Formatter interface {
 //
 // It's not exported because it's still using Data in an opinionated way. It's to
 // avoid code duplication between the two default formatters.
-func prefixFieldClashes(data Fields) {
+func prefixFieldClashes(data Fields, reportCaller bool) {
```

Comment:
fieldKey only exists in the JSON formatter right now, but even if that was ported into the text formatter the Fields passed in don't include whether the calling function will be added later - it just has the fields passed via WithField(s).  

This remapping actually looks like it may have issues if the field-name override is used, since the protection added is only for the original names and not the overrides.  I'd be open to looking at bringing the field mapping into the text formatter and then making prefixFieldClashes() handle this case, but it doesn't seem like that should be conflated with my current change.

The condition here was to ensure that if current users are making use of the field we use for the calling function (now 'func') and are not setting ReportCaller, then their logging isn't changed.  Refactors to accomplish this without adding the boolean all seemed much more more significant, and it didn't seem that they were justified.  If you see something I'm missing, please call it out.